onnxruntime/core/providers/cpu/quantization/quantize_linear.cc

```diff
@@ -439,6 +529,72 @@ void ComputeLoop(OpKernelContext* ctx, const InT* input, const InT* scale, const
   }
 }
 
+// Quantizes float32 to INT4 (in-place) using MLAS kernel.
+#define DEFINE_COMPUTE_LOOP_FP32_TO_INT4(INT4_TYPE, QUANT_FUNC)                                               \
+  template <>                                                                                                 \
+  void ComputeLoop(OpKernelContext* ctx, const float* input, const float* scale, const INT4_TYPE* zero_point, \
+                   INT4_TYPE* output, int64_t N, int64_t broadcast_dim, int64_t block_size, bool saturate) {  \
+    ORT_UNUSED_PARAMETER(saturate);                                                                           \
+    size_t output_index = 0;                                                                                  \
+    for (size_t n = 0; n < static_cast<size_t>(N); n++) {                                                     \
+      for (size_t bd = 0; bd < static_cast<size_t>(broadcast_dim); bd++) {                                    \
+        size_t bd_i = bd >> 1;  /*bd / 2*/                                                                    \
+        size_t bd_j = bd & 0x1; /*bd % 2*/                                                                    \
+        INT4_TYPE::UnpackedType zp = zero_point ? zero_point[bd_i].GetElem(bd_j) : 0;                         \
+        QUANT_FUNC(input, output, output_index, output_index + static_cast<size_t>(block_size),               \
+                   scale[bd], INT4_TYPE(zp, 0), ctx->GetOperatorThreadPool());                                \
+        input += block_size;                                                                                  \
+        output_index += static_cast<size_t>(block_size);                                                      \
+      }                                                                                                       \
+    }                                                                                                         \
+    assert(output_index == static_cast<size_t>(N * broadcast_dim * block_size));                              \
+  }
```

Comment:
the loop is consuming block_size. May I know what's the reason of not allowing blockSize in the op attribute?

Reply:
Although this variable is called "block_size", it is not the same thing as the new block_size attribute.

Reply:
The new block_size attribute should essentially split the `broadcast_dim` into chunks. The current `block_size` variable needs to be renamed to avoid confusion.